graphql_compiler/compiler/emit_sql.py

```diff
@@ -590,21 +629,25 @@ def add_filter(self, predicate, aliases):
         sql_expression = predicate.to_sql(self._dialect, aliases, self._output_vertex_alias)
         self._filters.append(sql_expression)
 
-    def end_fold(self, alias_generator, from_clause, outer_from_table):
+    def end_fold(
+        self,
+        alias_generator: "UniqueAliasGenerator",
```

Comment:
Would it be feasible to move the definition of `UniqueAliasGenerator` above this class, to avoid needing this forward reference? Happy to wait for a separate PR if so, no need to make this one more complex.

Reply:
@obi1kenobi see if you agree with intent of this upcoming PR https://github.com/kensho-technologies/graphql-compiler/pull/813 since it makes this irrelevant

Reply:
Sounds good to me, happy with #813 and also generally happy to delegate such decisions to you two.